sequencer/src/block.rs

```diff
@@ -17,28 +19,131 @@ use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use std::{
     env,
     fmt::{Debug, Display},
+    ops::Add,
     time::Duration,
 };
 use time::OffsetDateTime;
-use typenum::U3;
+use typenum::Unsigned;
+use typenum::U2;
 
 pub type BlockMerkleTree = LightWeightSHA3MerkleTree<Commitment<Header>>;
 pub type BlockMerkleCommitment = <BlockMerkleTree as MerkleTreeScheme>::Commitment;
 
-#[derive(Default)]
+// New Type for `U256` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct FeeElement(U256);
+// New Type for `Address` in order to implement `CanonicalSerialize` and
+// `CanonicalDeserialize`
+#[derive(
+    Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord,
+)]
+pub struct FeeIndex(Address);
```

Comment:
Nit: the `Index` here refers to only one usage of this type, where we use it as an index in a key-value map. But we may well end up using this type elsewhere, any time we deal with fees. WDYT about naming this something more descriptive like `FeeAccount` or `FeeAddress`?